### PR TITLE
Adding OpenAPI in the Reactive Routes Guide

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR adds the OpenAPI section to the Reactive Routes Guide. (see PR #10944)

It also updates to the latest SmallRye OpenAPI version that contains some more fixes for the Reactive Routes, namely
- Support for default method on a Route (GET)
- Support for default path on a Route (Method Name)

### CODE REVIEW

**Comment by mkouba** [comment]
> Support for default method on a Route (GET)

Hm, if a route specifies no method then all methods are assumed. This is the default behavior from vertx-web.

**Comment by phillip-kruger** [comment]
Ok, can I add the others in the next version of smallrye ? I'll open an issues there

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 1]
```diff
@@ -522,6 +522,171 @@ public class MyFilters {
...
+          content:
+            '*/*':
+              schema:
+                type: string
+----
+
+=== Using Swagger UI
+
+Swagger UI is included by default when running in `Dev` or `Test` mode, and can optionally added to `Prod` mode.
+See link:https://quarkus.io/guides/openapi-swaggerui#use-swagger-ui-for-development[the Swagger UI] Guide for more details.
```
Comment (mkouba) [inline docs/src/main/asciidoc/reactive-routes.adoc:684]: Is this the correct way to navigate to a section in another document? CC @gsmet

The URL part is probably pretty stable but I think that `<<openapi-swaggerui.adoc#use-swagger-ui-for-development,the Swagger UI>>` should work as well.

Comment (phillip-kruger) [reply docs/src/main/asciidoc/reactive-routes.adoc:684]: Ok let me change that.

Comment (phillip-kruger) [reply docs/src/main/asciidoc/reactive-routes.adoc:684]: mmm, the rest of the document use full URLs (see line 78 - 87) - should I change all of them. Or stay as is to be consistent ?

Comment (phillip-kruger) [reply docs/src/main/asciidoc/reactive-routes.adoc:684]: Forget that. Those are different domains. I am changing.

**Review by mkouba — APPROVED** [review]
LGTM.

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 2]
```diff
@@ -522,6 +522,171 @@ public class MyFilters {
 <1> The `RouteFilter#value()` defines the priority used to sort the filters - filters with higher priority are called first.
 <2> The filter is likely required to call the `next()` method to continue the chain.
 
+== Adding OpenAPI and Swagger-UI
+
+You can add support for link:https://www.openapis.org/[OpenAPI] and link:https://swagger.io/tools/swagger-ui/[Swagger-UI] by using the `quarkus-smallrye-openapi` extension.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:527]:
```suggestion
You can add support for link:https://www.openapis.org/[OpenAPI] and link:https://swagger.io/tools/swagger-ui/[Swagger UI] by using the `quarkus-smallrye-openapi` extension.
```

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 3]
```diff
@@ -522,6 +522,171 @@ public class MyFilters {
 <1> The `RouteFilter#value()` defines the priority used to sort the filters - filters with higher priority are called first.
 <2> The filter is likely required to call the `next()` method to continue the chain.
 
+== Adding OpenAPI and Swagger-UI
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:525]:
```suggestion
== Adding OpenAPI and Swagger UI
```

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 4]
```diff
@@ -522,6 +522,171 @@ public class MyFilters {
...
+          description: OK
+          content:
+            '*/*':
+              schema:
+                type: string
+----
+
+=== Using Swagger UI
+
+Swagger UI is included by default when running in `Dev` or `Test` mode, and can optionally added to `Prod` mode.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:683]:
```suggestion
Swagger UI is included by default when running in `dev` or `test` mode, and can optionally added to `prod` mode.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Added some very minor suggestions. If you can apply them and squash, that would be great.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**